python/hsfs/feature_group.py

```diff
@@ -2084,12 +2084,24 @@ def read(
         # Raises
             `hsfs.client.exceptions.RestAPIError`. No data is available for feature group with this commit date, If time travel enabled.
         """
+        if wallclock_time and self._time_travel_format is None:
+            raise FeatureStoreException(
+                "Time travel format is not set for the feature group, cannot read as of specific point in time."
+            )
+        elif wallclock_time and engine.get_type() == "python":
```

Comment:
hive and flyduck does not support `as_of` right?

Reply:
Nope